src/store/getters.ts

```diff
@@ -9,3 +32,355 @@ export const getNameRequestUrl = (state: any): string => {
+      }
+    } else if (this.request_action_cd === 'MVE' && ['CA', 'IN'].includes(getLocation(state))) {
+      // Shortlist order: Limited company, Cooperative association
+      if (x.value === 'XLP') {
+        x.shortlist = null
+        x.rank = null
+      } else if (x.value === 'XCP') {
+        x.shortlist = true
+        x.rank = 2
+      }
+    } else if (this.request_action_cd === 'CHG' && ['CA', 'IN'].includes(getLocation(state))) {
+      // Shortlist order: Limited company, Limited partnership, Limited liability partnership
+      if (x.value === 'XLP') {
+        x.shortlist = true
+        x.rank = 2
+      } else if (x.value === 'XLL') {
+        x.shortlist = true
+        x.rank = 3
+      } else if (x.value === 'XCP') {
+        x.shortlist = null
+        x.rank = null
+      }
+    }
+    if (x.shortlist) {
+      return x
+    }
+  })
+  let options: SelectOptionsI[] = getLocation(state) === 'BC' ? [...bcOptions] : [...xproOptions]
+  let n = 4
+
+  if (this.entityTypeAddToSelect) {
+    this.entityTypeAddToSelect.rank = 4
+    options = options.concat(this.entityTypeAddToSelect)
+    n = 5
+  }
+  options = options.concat({ text: 'View all business types', value: 'INFO', rank: n })
+  return options.sort((a, b) => {
+    if (a.rank < b.rank) {
+      return -1
+    }
+    if (a.rank > b.rank) {
+      return 1
+    }
+    return 0
+  })
+}
+
+/** Return true if the name is Slashed */
+export const getNameIsSlashed = (state: StateIF) => {
+  if (getName(state)) {
+    let name = getName(state)
+    if (name.includes('/') && name.split('/').length === 2) {
+      name = name.replace(/(\s+|(?=.))\/(\s+|(?=.))/g, '/')
+      let leftSideWords = name.split('/')[0].split(' ')
+      let rightSideWords = name.split('/')[1].split(' ')
+      if (leftSideWords.length >= 2 && rightSideWords.length >= 2) {
+        return true
+      }
+    }
+  }
+  return false
+}
+
+export const getShowCorpNum = (state: StateIF): 'colin' | 'mras' | false => {
+  if (($colinRequestActions.includes(this.request_action_cd) && $colinRequestTypes.includes(getEntityTypeCd(state))) ||
+    this.entity_type_cd === 'DBA') {
+    return 'colin'
+  }
+  if ($colinRequestActions.includes(this.request_action_cd) &&
+    $xproColinRequestTypes.includes(getEntityTypeCd(state))) {
+    return 'colin'
+  }
+  let mrasEntities = ['XCR', 'XLP', 'UL', 'CR', 'CP', 'BC', 'CC']
+  let { xproJurisdiction } = this.nrData
+
+  if ($mrasJurisdictions.includes(xproJurisdiction?.toLowerCase()) && mrasEntities.includes(getEntityTypeCd(state))) {
+    if (getLocation(state) === 'CA' && ['NEW', 'ASSUMED'].includes(getRequestActionCd(state))) {
+      return 'mras'
+    }
+    if (getLocation(state) === 'BC' && ['MVE'].includes(getRequestActionCd(state))) {
+      return 'mras'
+    }
+  }
+  return false
+}
+
+// MODAL GETTERS
+export const getIncorporateLoginModalVisible = (state: StateIF): boolean => {
+  return state.stateModel.newRequestModel.incorporateLoginModalVisible
```

Comment:
FYI, component visibility state I hope to refactor out. 🙄

Reply:
Would be nice to have a singular action/mutation with params for all modal visibility states, but it sounds like we will be going another route. Using v-model maybe? What are you thinking?

Reply:
I've only made a todo list step so far 😁 

I don't have a solution yet.